File: packages/ruviz-web/examples/heatmap.ts
import { createPlot } from "../src/index.js";
import { heatmapValues } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "heatmap",
  title: "Heatmap",
  summary: "A rectangular numeric matrix rendered as a heatmap.",
  category: "matrix",
  createPlot() {
    return createPlot().sizePx(640, 320).theme("dark").title("Heatmap").heatmap(heatmapValues());
  },
};

File: python/examples/heatmap.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, heatmap_values, save_example

META = ExampleMeta(
    slug="heatmap",
    title="Heatmap",
    summary="A rectangular numeric matrix rendered as a heatmap.",
    section="Matrix plots",
)


def build_plot():
    return base_plot("Heatmap", theme="dark").heatmap(heatmap_values())


if __name__ == "__main__":
    save_example(META, build_plot())
